polliwog/polyline/_polyline_object.py

```diff
@@ -301,6 +301,16 @@ def aligned_with(self, vector):
         else:
             return self
 
+    def aligned_using_points(self, p1: np.ndarray, p2: np.ndarray):
+        """
+        Return a new Polyline which is conditionally flipped so that p1 is
+        sequentially before p2.
+        """
+        return self.flipped_if(
```

Comment:
Paul commented: "I think this approach returns something reasonable for open polylines but not closed, since the index doesn't really matter and it's more the distance we'd need to consider"

Reply:
This is still useful to call on closed polylines, when your are slicing it by points afterwards.  This ensures that the correct segment is returned.

Reply:
Makes sense about the problem you're solving, though I think that use case isn't solved by combining these two methods. I think you will get different behavior here depending when v[0] happens to fall between p1 and p2. In that case it'll get flipped and I think you'd get the wrong segment (albeit in the right order).